File: day19/main_test.go
package main

import (
	_ "embed"
	"testing"

	"github.com/stretchr/testify/assert"
)

//go:embed ex
var testEx string

//go:embed inp
var testInp string

func Test_Part1Ex(t *testing.T) {
	assert.Equal(t, 6, new(solver).prep(testEx).part1())
}

func Test_Part1(t *testing.T) {
	assert.Equal(t, 0, new(solver).prep(testInp).part1())
}

func Test_Part2Ex(t *testing.T) {
	assert.Equal(t, 0, new(solver).prep(testEx).part2())
}

func Test_Part2(t *testing.T) {
	assert.Equal(t, 0, new(solver).prep(testInp).part2())
}

File: day19/src/main_test.rs
#[cfg(test)]
mod tests {
    use crate::Solver;
    use std::fs;

    #[test]
    fn part1_example() {
        let content = fs::read_to_string("./ex");
        assert!(content.is_ok());

        let s: Solver = Solver::new().prep(content.unwrap());
        assert_eq!(6, s.part1());
    }

    #[test]
    fn part1_inp() {
        let content = fs::read_to_string("./inp");
        assert!(content.is_ok());

        let s: Solver = Solver::new().prep(content.unwrap());
        assert_eq!(336, s.part1());
    }

    #[test]
    fn part2_example() {
        let content = fs::read_to_string("./ex");
        assert!(content.is_ok());

        let s: Solver = Solver::new().prep(content.unwrap());
        assert_eq!(16, s.part2());
    }

    #[test]
    fn part2_inp() {
        let content = fs::read_to_string("./inp");
        assert!(content.is_ok());

        let s: Solver = Solver::new().prep(content.unwrap());
        assert_eq!(758890600222015, s.part2());
    }
}
